src/api.rs

```diff
@@ -229,30 +249,77 @@ impl Api {
         Ok(())
     }
 
-    fn register_input(&mut self, request: RegisterInputRequest) -> Result<(), ApiError> {
+    fn register_input(
+        &mut self,
+        request: RegisterInputRequest,
+    ) -> Result<ResponseHandler, ApiError> {
         let RegisterInputRequest { input_id: id, port } = request;
+        let port: Port = port.try_into()?;
 
-        if let Some((node_id, _)) = self.pipeline.inputs().find(|(_, input)| input.port == port) {
-            return Err(ApiError::new(
-                "PORT_ALREADY_IN_USE",
-                format!("Failed to register input stream \"{id}\". Port {port} is already used by node \"{node_id}\""),
-                tiny_http::StatusCode(400)
-            ));
-        }
+        match port {
+            Port::Range((start, end)) => {
+                for port in start..end {
+                    trace!("[input {}] checking port {}", id, port);
 
-        let result = self.pipeline.register_input(
-            id.clone().into(),
-            rtp_receiver::Options {
-                port,
-                input_id: id.into(),
-            },
-        );
+                    if self.pipeline.inputs().any(|(_, input)| input.port == port) {
+                        continue;
+                    }
```

Comment:
Test if FFmpeg reserves the second port as well. If it does, check for it here, to avoid checking it on potentially expensive `register_input`.